File: backend/src/main/java/com/patriot/fourlipsclover/plan/dto/request/PlanScheduleUpdateRequest.java
package com.patriot.fourlipsclover.plan.dto.request;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlanScheduleUpdateRequest {

    private Integer restaurantId;
    private String notes;      // 추가 메모
    private LocalDateTime visitAt;  // 변경된 방문 예정 일시

}

File: frontend/lib/models/plan/plan_schedule_update_request.dart
class PlanScheduleUpdateRequest {
  final int? restaurantId;
  final String? kakaoPlaceId; // 카카오 Place ID 추가
  final String? notes;
  final DateTime visitAt;

  PlanScheduleUpdateRequest({
    this.restaurantId,
    this.kakaoPlaceId,
    this.notes,
    required this.visitAt,
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {
      'visitAt': visitAt.toIso8601String(),
    };
    
    if (restaurantId != null) {
      data['restaurantId'] = restaurantId;
    }
    
    if (kakaoPlaceId != null) {
      data['kakaoPlaceId'] = kakaoPlaceId;
    }
    
    if (notes != null) {
      data['notes'] = notes;
    }
    
    return data;
  }
}